gnomad/utils/sparse_mt.py

```diff
@@ -344,29 +406,34 @@ def get_as_info_expr(
         logger.error(msg)
         raise ValueError(msg)
 
-    # Modify aggregations to aggregate per allele
-    agg_expr = {
-        f: hl.agg.array_agg(
-            lambda ai: hl.agg.filter(mt.LA.contains(ai), expr),
-            mt[alt_alleles_range_array_field],
-        )
-        for f, expr in agg_expr.items()
-    }
+    if not treat_fields_as_allele_specific:
+        # Modify aggregations to aggregate per allele
+        agg_expr = {
+            f: hl.agg.array_agg(
+                lambda ai: hl.agg.filter(mt.LA.contains(ai), expr),
+                mt[alt_alleles_range_array_field],
+            )
+            for f, expr in agg_expr.items()
+        }
 
     # Run aggregations
     info = hl.struct(**agg_expr)
 
-    # Add SB Ax2 aggregation logic and FS if SB is present
-    if "AS_SB_TABLE" in info:
-        as_sb_table = hl.array(
-            [
-                info.AS_SB_TABLE.filter(lambda x: hl.is_defined(x)).fold(
-                    lambda i, j: i[:2] + j[:2], [0, 0]
-                )  # ref
-            ]
-        ).extend(
-            info.AS_SB_TABLE.map(lambda x: x[2:])  # each alt
-        )
+    # Add FS and SOR if SB is present.
+    if "AS_SB_TABLE" in info or "AS_SB" in info:
+        # Rename AS_SB to AS_SB_TABLE if present and add SB Ax2 aggregation logic.
+        if "AS_SB" in agg_expr:
```

Comment:
Not sure this will happen but do we always want to overwrite AS_SB_TABLE with AS_SB when both are present? Should we have logic that throws a warning if this happens? Or is that the intended use?